export-osv.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * Script for exporting advisories to OSV format.
+ *
+ * Usage: `php export-osv.php target_folder`
+ *
+ * @see https://ossf.github.io/osv-schema/
+ */
+
+namespace FriendsOfPhp\SecurityAdvisories;
+
+use DirectoryIterator;
+use FilesystemIterator;
+use SplFileInfo;
+use Symfony\Component\Yaml\Yaml;
+
+if (!is_file($autoloader = __DIR__ . '/vendor/autoload.php')) {
+    echo 'Dependencies are not installed, please run "composer install" first!' . PHP_EOL;
+    exit(1);
+}
+
+require $autoloader;
+
+function convertToOsv(SplFileInfo $fileInfo, string $package): array
+{
+    $advisory = Yaml::parseFile($fileInfo->getPathname());
+
+    return [
+        'id' => $advisory['cve'] ?? 'PHPSEC-' . $fileInfo->getBasename('.yaml'),
+        'modified' => getDateFromGitLog($fileInfo),
+        'published' => getDateFromGitLog($fileInfo, true),
+        'aliases' => [],
+        'related' => [],
+        'summary' => $advisory['title'] ?? '',
+        'details' => '',
+        'affected' => [
+            'package' => [
+                'ecosystem' => 'Packagist',
+                'name' => $package,
+                'purl' => sprintf('pkg:packagist/%s', $package),
+            ],
+            'ranges' => [
+                'type' => 'SEMVER',
+                'events' => getEvents($advisory['branches']),
+            ],
+        ],
+        'references' => [
+            array_key_exists('link', $advisory) ? [
+                'type' => 'ADVISORY',
+                'url' => $advisory['link'],
+            ] : null,
+            [
+                'type' => 'PACKAGE',
+                'url' => 'https://packagist.org/packages/' . $package,
+            ],
+        ],
+    ];
+}
+
+function getEvents(array $branches): array
+{
+    $events = [];
+
+    foreach (array_column($branches, 'versions') as $branch) {
+        if (count($branch) === 2) {
+            array_push($events, ['introduced' => $branch[0]]);
+            array_push($events, ['fixed' => $branch[1]]);
```

Comment:
be careful. We have cases like `>= 1.3.0, <2.0` where things are not fixed in 2.0 (if another branch has `>=2.0` in its affected constraints)

Reply:
That's true. I reworked it, see https://github.com/FriendsOfPHP/security-advisories/pull/599#discussion_r752573875.